Datashuttle: Deploy Datashuttle as an installable package (Lucas Laredo)

laredoo opened the pull request:
## Description

Hi everyone! This is my project proposal for GSoC'25. I am available for discussion and look forward to your feedback and suggestions.

I politely ask @JoeZiminski and @adamltyson to review this PR and if I'm missing something, please let me know! Otherwise, if everything is in place, I can submit it in the GSoC official platform :)

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [x] Other

**Why is this PR needed?**

This is the required PR to present my proposal for the Datashuttle project. It's mandatory in order to apply to GSoc 2025.

**What does this PR do?**

This pull request provides the filled application template for my project.

adamltyson commented on GSoC-2025/application_template.md:
Could you restore this file? Just so it isn't deleted when we merge the PR.

adamltyson commented on GSoC-2025/GSoC-2025/Datashuttle: Deploy Datashuttle as an installable package (Lucas Laredo).md:
I'm not sure what "project standard packages" means here?

adamltyson commented on GSoC-2025/GSoC-2025/Datashuttle: Deploy Datashuttle as an installable package (Lucas Laredo).md:
What does "Implement Python API support" mean here?

adamltyson commented on GSoC-2025/GSoC-2025/Datashuttle: Deploy Datashuttle as an installable package (Lucas Laredo).md:
Does this include automatic building of the app, and release (e.g. via GitHub)?

adamltyson commented on GSoC-2025/GSoC-2025/Datashuttle: Deploy Datashuttle as an installable package (Lucas Laredo).md:
How many hours a week is your internship? Is it feasible to do both properly? Bear in mind that you plan for a longer GSoC project if that is helpful.

laredoo replied:
Thank you for all the suggestions, @adamltyson! I'll make sure to address all of them, but since this might impact the implementation timeline, I think it's better to start from here. Can I propose a longer GSoC project (more than 12 weeks, for example)? The internship is 20h/week, and since I'll be on vacation from University I'm confident that I can balance both. Let me know what you think!

adamltyson reviewed (commented):
Hey @laredoo. This is a nice proposal, I've left some small specific suggestions. More generally though, it would be good to make the proposal a bit more specific. What technologies would you consider and why? Do you envision any potential issues?

laredoo commented:
Hey @adamltyson, thanks again for your feedback! I’ve addressed all your suggestions and refined the implementation timeline to make the proposal much more specific. I also added two new columns—one for expected results and another for notes—to better track progress each week. Let me know if there’s anything else I should clarify!

adamltyson commented:
Hi @laredoo, If you've addressed the comments then I think you're ready to submit. Please bear in mind the page limits though, as your application seems to have got considerably longer (I appreciate it's hard to fit everything in!). Apologies that we're not able to give too much more feedback at this stage.

laredoo commented:
> Hi @laredoo, If you've addressed the comments then I think you're ready to submit. Please bear in mind the page limits though, as your application seems to have got considerably longer (I appreciate it's hard to fit everything in!). Apologies that we're not able to give too much more feedback at this stage.

Hey @adamltyson, thanks a lot for all your help!  I've revised the proposal to stay within the page limits and made sure to address all your comments. Also, I rebased this branch to sort out the merge conflicts, so everything should be good to go now. And no worries at all about the level of feedback — what you shared was already super helpful and really appreciated. Thanks again!
